src/core/imap.ts

```diff
@@ -0,0 +1,117 @@
+import * as Imap from 'imap';
+import {ImapConfig} from '../model/imapConfig.model';
+import {Query} from '../model/Query.model';
+
+export class IMAP {
+    private imap = null;
+    private monitorId = null;
+    private query: Query = null;
+    private delayTime = 1000 * 5;
+
+    constructor(config: ImapConfig) {
+        this.imap = new Imap({
+            ...config,
+            authTimeout: 10000,
+            connTimeout: 30000,
+            tlsOptions: {
+                rejectUnauthorized: false,
+            },
+
+        });
+    }
+
+    init() {
+        return new Promise((resolve, reject) => {
+            this.imap.connect();
+
+            this.imap.once('ready', ()=> {
+                console.log('%c---------------------  SUCCESSFUL IMAP CONNECTION  --------------------', 'color: Green');
+                resolve(this.imap);
+            });
+
+            this.imap.once('error', function(err: Error) {
+                reject(err);
+            });
+        });
+    }
+
+    setQuery(query: Query) {
+        this.query = query;
+    }
+
+    openBox(mailBox, readOnly = true) {
+        return new Promise((resolve, reject) => {
+            this.imap.openBox(mailBox, readOnly, (err, box) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(box);
+                }
+            });
+        });
+    }
+
+    search(criteria) {
+        return new Promise((resolve, reject) => {
+            this.imap.search(criteria, (err, messages) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(messages);
+                }
+            });
+        });
+    }
+
+    state() {
+        return new Promise(async (resolve, reject) => {
+            if (!navigator.onLine) {
+                reject(new Error('No internet Connection'));
+            } else if (!this.imap) {
+                reject(new Error('Please Re-login'));
+            } else if (this.imap.state === 'authenticated') {
+                resolve('authenticated');
+            } else {
+                try {
+                    await this.init();
+                    resolve('Reconnected successfully');
```

Comment:
You resolve this passing string but never use the return value.

Reply:
I think this is where having type annotations could help, then it'd be immediately clear that you don't use the return type.

Reply:
> You resolve this passing string but never use the return value.

I was trying to make the function more readable without comments.

Reply:
> I think this is where having type annotations could help, then it'd be immediately clear that you don't use the return type.

Yes, I annotated the promises by void type.